data_safe_haven/config/backend_settings.py

```diff
@@ -0,0 +1,144 @@
+"""Load global and local settings from dotfiles"""
+# Standard library imports
+import pathlib
+from typing import Optional
+
+# Third party imports
+import appdirs
+import yaml
+
+# Local imports
+from data_safe_haven.exceptions import DataSafeHavenParameterException
+from data_safe_haven.utility import Logger
+
+
+class BackendSettings:
+    """Load global and local settings from dotfiles with structure like the following
+
+    azure:
+      admin_group_id: d5c5c439-1115-4cb6-ab50-b8e547b6c8dd
+      location: uksouth
+      subscription_name: Data Safe Haven Development
+    current:
+      name: Turing Development
+    """
+
+    def __init__(
+        self,
+    ) -> None:
+        # Define instance variables
+        self._admin_group_id: Optional[str] = None
+        self._location: Optional[str] = None
+        self._name: Optional[str] = None
+        self._subscription_name: Optional[str] = None
+        self.logger = Logger()
+
+        # Load previous backend settings (if any)
+        self.config_directory = pathlib.Path(
+            appdirs.user_config_dir("data_safe_haven")
+        ).resolve()
+        self.config_file_path = self.config_directory / ".dshconfig"
```

Comment:
I'm confused about what you're suggesting here. At some point we need to set this name from a string. Would you rather see something like this?

```
config_file_name = "config.yaml"
self.config_file_path = self.config_directory / config_file_name
```

Reply:
Suggestion was to put the name at class or module level in case it's needed later. For the moment it's not, so we'll leave it. May be refactored in future.